wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wom2wdlom/WomToWdlomImpl.scala

```diff
@@ -0,0 +1,311 @@
+    new WomToWdlom[InputDefinition, InputDeclarationElement] {
+      override def toWdlom(a: InputDefinition): InputDeclarationElement = a match {
+        case a: RequiredInputDefinition => InputDeclarationElement(a.womType.toWdlom, a.localName.value, None)
+        case a: InputDefinitionWithDefault => InputDeclarationElement(a.womType.toWdlom, a.localName.value, Some(a.default.toWdlom))
+        case a: FixedInputDefinition => InputDeclarationElement(a.womType.toWdlom, a.localName.value, Some(a.default.toWdlom))
+        case a: OptionalInputDefinition => InputDeclarationElement(a.womType.toWdlom, a.localName.value, None)
+      }
+    }
+
+  implicit val callableTaskDefinitionToTaskDefinitionElement: WomToWdlom[CallableTaskDefinition, TaskDefinitionElement] =
+    new WomToWdlom[CallableTaskDefinition, TaskDefinitionElement] {
+      override def toWdlom(a: CallableTaskDefinition): TaskDefinitionElement = {
+        // TODO: why does _.toWdlom not work here?
+        val inputs = a.inputs.map(inputDefinitionToInputDeclarationElement.toWdlom)
+        val outputs = a.outputs.map(outputDefinitionToOutputDeclarationElement.toWdlom)
+
+        val command = a.commandTemplateBuilder(Map()).getOrElse(???)
+
+        val commandLine = CommandSectionLine(command map {
+          case s: StringCommandPart =>
+            StringCommandPartElement(s.literal)
+          case p: ParameterCommandPart =>
+            val attrs = PlaceholderAttributeSet(
+              defaultAttribute = p.attributes.get("default"),
+              trueAttribute = p.attributes.get("true"),
+              falseAttribute = p.attributes.get("false"),
+              sepAttribute = p.attributes.get("sep")
+            )
+
+            // TODO: it can't be anything but a Terminal, but this is not the best way to express that
+            import wdl.draft2.parser.WdlParser.Terminal
+            PlaceholderCommandPartElement(ExpressionLiteralElement(p.expression.ast.asInstanceOf[Terminal].getSourceString), attrs)
+        })
+
+        TaskDefinitionElement(
+          a.name,
+          if (inputs.nonEmpty) Some(InputsSectionElement(inputs)) else None,
+          Seq(), // TODO: maybe these don't exist in draft-2?
+          if (outputs.nonEmpty) Some(OutputsSectionElement(outputs)) else None,
+          CommandSectionElement(Seq(commandLine)),
+          a.runtimeAttributes.toWdlom,
+          mapToMetaSectionElement.toWdlom(a.meta), // TODO: why do these require explicit notation?
+          mapToParameterMetaSectionElement.toWdlom(a.parameterMeta)
+        )
+      }
+    }
+
+  implicit val workflowDefinitionToWorkflowDefinitionElement: WomToWdlom[WorkflowDefinition, WorkflowDefinitionElement] =
+    new WomToWdlom[WorkflowDefinition, WorkflowDefinitionElement] {
+      override def toWdlom(a: WorkflowDefinition): WorkflowDefinitionElement = {
+        val inputs = a.inputs.map(inputDefinitionToInputDeclarationElement.toWdlom)
```

Comment:
Side note: that was all thinking out loud about the future... but since what we really want is something that works (and this does work!), I think we should be not worrying about alternative structures for now, and consider refactoring it *only if we actually need to* going into 1.1